server.py

```diff
@@ -360,6 +366,49 @@ def healthcheck():
         return jsonify({'status': 'OK'})
 
 
+def get_all_comments_by_survey_id(survey_id):
+    # collect survey based on survey id and and 146 code
+
+    db.engine.echo = True
+    max_surveys = 10
+
+    # retrieve all the records
+    comments = list()
+    sliced_value = 0
+    while True:
+        survey_records_page = db.session.query(SurveyResponse).slice(sliced_value, max_surveys).all()
+        if not survey_records_page or len(survey_records_page) == 0:
+            break
+
+        for data in survey_records_page:
+            logger.debug(data.data)
+            record = json.loads(data.data)
+            if '146' in record['data']:
+                if survey_id in record['survey_id']:
+                    comments.append((record['survey_id'], record['data']['146']))
+        sliced_value = len(survey_records_page)
+    logger.info("Comments retrieved " + str(len(comments)))
+    return comments
+
+
+@app.route('/comments/<string:survey_id>', methods=['GET'])
+def get_comments(survey_id):
+    logger.info("Exporting comments for survey id " + survey_id)
+
+    if survey_id is None:
+        logger.error("Survey_id is none : " + survey_id)
+        return server_error(400)
+
+    logger.info("Exporting comments for survey id " + survey_id)
+    try:
+        workbook = exporter.create_comments_book(survey_id, get_all_comments_by_survey_id(survey_id))
+    except Exception as e:
+        logger.error("File generation went wrong ", error=e)
+        return server_error(500)
```

Comment:
probably better standard to be a `raise` instead of a return in an instance of 5xx

Reply:
I mean I am not sure 100%, i have tried to be complaiant with the other errors in the other routes.